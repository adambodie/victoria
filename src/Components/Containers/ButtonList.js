import React from 'react'
import Button from '../items/Button'

const ButtonList = ({data}) => (
	<div className='buttons'>
		{data.map((x, index) => index !== 0 && (<Button link={x.link} name={x.logo} key={index} />))}
	</div>
)

export default ButtonList
