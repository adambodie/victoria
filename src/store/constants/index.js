export const INCREMENT = 'INCREMENT';
export const DECREMENT = 'DECREMENT';

export const BUTTONS = [
    'ʔ', 'a', 'c', 'c̕', 'č', 'č̕','e', 'ə', 'h', 'i', 'k', 'kʷ', 'k̕ʷ', 'l', 
    'ɬ', 'ƛ̕', 'm', 'm̕', 'n', 'n̕', 'ŋ', 'ŋ̕','p', 'p̕', 'q', 'q̕', 'qʷ', 'q̕ʷ', 
    's', 'š', 't', 't̕', 'u', 'w', 'w̕', 'xʷ', 'x̣', 'x̣ʷ', 'y', 'y̕', 
    '?', '.', ',', '!', '"', '\'', 'Delete', 'Clear', 'Space'
    ]

export const PAGES = [
    {'title': 'Welcome to Port Angeles', 'begin': 0, 'end': 25}, 
    {'title': 'Port Angeles Before the Ferry Ride', 'begin': 26, 'end': 40}, 
    {'title': 'On the MV Coho, Ready to Go to Canada', 'begin': 41, 'end': 49}, 
    {'title': 'Strait of Juan de Fuca', 'begin': 50, 'end': 63}, 
    {'title': 'Juan de Fuca Strait', 'begin': 64, 'end': 87}, 
    {'title': 'Entering Victoria on a Perfect Day', 'begin': 88, 'end': 131}, 
    {'title': 'Exploring the Diversity of Victoria', 'begin': 132, 'end': 160}, 
    {'title': 'Beacon Hill Park: The Gem of The Garden City', 'begin': 161, 'end': 195}, 
    {'title': 'Walking Along the Canadian Coast', 'begin': 196, 'end': 205}, 
    {'title': 'Back in the Harbour to Explore', 'begin': 206, 'end': 239}, 
    {'title': 'Back on the Ferry: Back to the States', 'begin': 240, 'end': 268}, 
    {'title': 'Sequim, Blyn and the Hood Canal Bridge Closure', 'begin': 269, 'end': 287}, 
    {'title': 'One Last Unplanned Hurrah in Bremerton', 'begin': 288, 'end': 302}
]

export const LINKS = [
	{ 'link' : '/', 'name' : 'Home' },
    { 'link' : '/visit', 'logo' : 'camera', 'name' : 'Visit' },
    { 'link' : '/eat', 'logo' : 'cutlery', 'name' : 'Eat' },
    { 'link' : '/learn', 'logo' : 'graduation-cap', 'name' : 'Learn' },
    { 'link' : '/buy', 'logo' : 'usd', 'name' : 'Buy'},
    { 'link' : '/speak', 'logo' : 'bullhorn', 'name' : 'Speak'}
]

export const BANNERS = [
	{'name': 'coast-banner', 'alt' : 'Coastline'},
	{'name': 'harbour-banner', 'alt' : 'Inner Harbour'}, 
	{'name': 'parliament-banner', 'alt' : 'British Columbia Parliament Buildings'}, 
	{'name': 'port-angeles-banner', 'alt' : 'Port Angeles'}
]

export const FEATURES = [
	{
		'image': 'port-angeles',
		'alt': 'Port Angeles',
		'title': 'Port Angeles',
		'text': 'Visit the Gateway to the Olympic National Park'
	},
	{
		'image': 'coho',
		'alt': 'MV Coho',
		'title': 'MV Coho',
		'text': 'Take the Ferry from Port Angeles to Victoria'
	},
	{
		'image': 'chowder',
		'alt': 'Cuisine',
		'title': 'Cuisine',
		'text': 'Enjoy some fine Pacific Northwest cuisine'
	},
	{
		'image': 'klallam',
		'alt': 'Native Americans',
		'title': 'Native Americans',
		'text': 'Learn about the Native American Culture of the Pacific Northwest'
	},
	{
		'image': 'victoria',
		'alt': 'Victoria',
		'title': 'Victoria',
		'text': 'Enjoy the Sights of \'The Garden City\''
	},
	{
		'image': 'rainbow',
		'alt': 'Amazing',
		'title': 'Amazing',
		'text': 'Witness something truly amazing'
	}
]

export const RESTAURANTS = [{
	'title': 'Frugal\'s',
	'image': 'frugal',
	'address': [
		'1520 E Front Street',
		'Port Angeles, WA 98363'
	],
	'paragraph': [
		'Not wanting to stay out too late for dinner and ensuring I\'d be rested enough for the journey to Victoria, I found Frugal\'s, a fast food burger joint that proclaims itself to be the Best of the Northwest.  Located next door to the Days Inn I stayed at, I headed there on foot, only to realize it was only a drive-thru, so I walked back and hopped in my car.',
		'With two sides of the drive thru, even at 7:00 pm, there were plenty of cars in line.  Wanting something juicy, I got myself a burger in fries, and though I entered the left hand side of the drive thru, not realizing it would be a reach to grab my food on the opposite side of me, I was able to easily get my burger and fries, at a reasonable price of $5.96.  As I drove off, I was greeted with a Thank You sign in English, Clallam and another language (which also appeared to be Native American).',
		'After a hop skip of a drive back to my hotel room, I opened my burger and fries, enjoying each an every bite of it.  The burger was delicious and juicy, and along with the tasty french fries, it was the perfect meal to enjoy in your hotel room and watch basketball on the TV.  I also brought a little Pendleton Whiskey from home, which complemented the meal.  In all, a perfect easy meal.'
	]
}, {
	'title': 'Frankie\'s Modern Diner',
	'image': 'frankie',
	'address': [
		'910 Government St #38',
		'Victoria, BC V8W 1X3, Canada'
	],
	'paragraph': [
		'Located in downtown Victoria, I thought it would be a fun place to eat while researching online, so I had my lunch here.  I was expecting more of a Retro Cafe, the type where oldies jams would play on the jukebox.  I could envision my newly discovered Grandpa Buddy doing his best Fonz impression.  Entering the place, as soon as I walked up the stairs, I soon realized I would not be having that type of experience.  Instead, I found a nice friendly eatery with plenty of food options, drinks and all the pleasentries you would expect in Canada.',
		'Wanting to try a Canadian beer, not the kind you\'d find stacked with the Bud and Coors of the world, I opted for a Hoyne Pilsner, from a local craft brewery, to have something different.  A crisp summer beer seemed perfect for the type of weather outside.  Though the glass was frozen, the beer itself had a nice refreshing taste.  Sticking to wanting to have something different, I had a bowl of Chowder, which was presented nicely and tasted well also.',
		'When it was all said and done, I enjoyed my first Canadian meal, all at the price of 21.50 dollars, which of course, was its value in Canadian dollars.  After entering my debit card into a machine and answering a few questions on the handheld device, which was a little confusing due to its United States related questions, I got my receipt, which kindly told me that I only spent about 17.00 US Dollars.'
	]
}, {
	'title': 'Coyote BBQ Pub',
	'image': 'coyote',
	'address': [
		'201 E Front St',
		'Port Angeles, WA 98362'
	],
	'paragraph': [
		'Making my way back to Port Angeles, tired from a long and fun day in Victoria, I again opted for a different food experience, finding a BBQ restaurant named Coyotes.  The building itself was kind of ambiguous, which made it not stand out particularly, but once I walked inside, I felt like I was in the Wild West.',
		'Starting with a beverage, I chose the classic but not a drink I usually have, a Sidecar.  Noted with having Remy Martin cognac, the cocktail was tasty.',
		'For dinner, I again opted for something less traditional than my usual food, so I ordered a Nacho plate with Chili.  The chili, according to the server, was a new recipe. With plenty of meats, beans and spices, it was delicious, which I assured the server it was.  It complemented the Nachos well, which contained cheese, salsa and jalapeños, among other things. The only downside was that I wasn’t able to finish the whole plate, stopping having eaten a good 90%.'

	]
}, {
	'title': 'Horse & Cow Bar & Grill',
	'image': 'horse-cow',
	'address': [
		'536 4th St',
		'Bremerton, WA 98337'
	],
	'paragraph': [
		'With an unexpected trip to Bremerton due to my Waze App suggesting I take the route through Kitsap County rather than US 101 and my desire to see something new and exciting, my plans for the day were completely spontaneous.  Even visiting Bremerton was completely unplanned; only after browing through my phone I settled on visiting the Bremerton Boardwalk.',
		'Unsure of where I wanted to eat, and not wanting to walk so much, my first two ideas went south; one place that looked cool was closed on Sundays, another place looked to classy for how I was feeling.  At least the seagulls were able to get a table.  Walking around the area near my car, I settled for a place called the Horse & Cow Bar & Grill, as I didn\'t want to spend too much time walking.',
		'Entering the Horse & Cow, I noticed the funky and gritty vibe.  Not super caring of where I sat, I entered the bar and ordered myself a Georgetown Brewing Bodhizafa IPA.  Gotta love that name.  Not the best IPA I\'ve had but nothing wrong with trying a new Seattle beer.  Again, trying to be adventurous and taking advantage of their Sunday Special, I ordered a $6.99 Steak and Fries.  It had been years since I had a steak.  The steak was satisfying, though not particularly superb, I had to teach myself how to properly cut and eat my steak.  I felt a little off before eating, but after lunch, I felt a lot better.  Mission accomplished.'
	]
}]

export const FACTS = [
    'Hall of Fame Football player John Elway was born in Port Angeles', 
    'Congregation Emanu-El is the oldest Jewish Synagogue in Canada',
    'The population of Port Angeles is about 19,500',
    'Totem poles are mostly made to commemorate ancestors, mythology and other positive events regarding the culture, but a few of them have been built to bring shame to those who have wronged said tribe that built them.',
    'The United States and Canada have a maritime boundary dispute over the seaward boundary west of the mouth of the Strait of Juan de Fuca',
    'Port Angeles is the most populated city on the Olympic Peninsula, and home of the headquarters of Olympic National Park, which consists of the majority of the mountain range in the peninsula.',
    'Port Angeles is anglicized from its original name, Puerto de Nuestra de los Angeles (Port of Our Lady of the Angels), named by Spanish explorer Francisco de Eliza in 1791',
    'Victoria\'s Chinatown is the second oldest Chinatown in North America, after San Francisco\'s',
    'Thunderbird is a legendary creature of many Native American cultures, specifically of those along the Pacific Northwest Coast',
    'In 2003, during a proposed construction project, human remains and artifacts were found in Port Angeles, which led to the discovery of the largest prehistoric Indian village and burial ground found in the United States, known as Tse-whit-zen.',
    'Vancouver Island, like the city (in British Columbia and in Washington), is named after the British Officer of Royal Navy, George Vancouver, who explored the Pacific Northwest coast regions in 1791-1795',
    'The Klallam native American tribe lived Port Angeles, whose historical presence led to the installment of street signs in the Klallam language, in addition to English.',
    'For many years, the Empress Hotel did not have a front entrance sign, which once led workers to put up a sign stating \'Anyone who doesn\'t know this is The Empress shouldn\'t be staying here.\'',
    'The MV Coho, which carries passengers and vehicles from Port Angeles to Victoria, takes 90 minutes, traveling 20 nautical miles at a speed of 15 knots (about 17 mph)',
    'Port Angeles sits in the Olympic Mountains rain shadows, receiving only 15-20 inches a rain per year, a sharp contrast to the 140-170 inches of rain per year the Olympic Mountains\'s Hoh Rainforest receives, considered the rainiest part of the continous United States',
    'Vancouver Island is 290 miles in length and 62 miles in width at its widest point',
    'Terry Fox, whose statue stands in Beacon Hill Park, attempted to run across Canada with an artifical leg to raise money for cancer awareness.  Starting in Newfoundland, he had to stop in Thunder Bay, Ontario due to declining health and later passed away.',
    'The MV Coho can hold up to 110 vehicles and 1,000 passengers',
    'Canada 150 is to honor the 150th Anniversary of the formation of the Canadian Confederation, in which the British colonies formed with Nova Scotia and New Brunswick to become Canada, which occurred on July 1, 1867',
    'Vancouver Island is one of the few areas of Canada that can grow subtropical Mediterranean crops such as olives and lemons',
    'Thunderbird Park, part of the Royal BC Museum, is home of totem poles, mostly from the Haida, Gitxsan and Kwakwaka\'wakw tribes.',
    'Due to its climate and location, Sequim, Washington is nicknamed the \'Lavender Capital of North America,\' cultivating more Lavender than anywhere outside of France',
    'Mile Zero of the Trans Canada Highway, which spans from British Columbia to Newfoundland & Labrador, begins in Victoria',
    'The Strait of Juan de Fuca, known in Canada as the Juan de Fuca Strait, which separates the Olympic Peninsula of Washington State from Vancouver Island of British Columbia, is about 96 miles long',
    'Juan de Fuca was a Greek navigator who claimed to be the first non-native to explore the strait, looking for a semi-mythical Strait of Anian that theoretically separated North America from Asia',
    'Vancouver Island is the largest island on the West Coast of North America, and the largest island in the Pacific Ocean east of New Zealand',
    'The southern part of Vancouver island is the only part of Western Canada below the 49th parallel',
    'Vancouver Island has a population of about 780,000, with about half of them living in Greater Victoria',
    'As of 2011, Aboriginal Canadians (including the First Nations tribes) consisted of about 6% of Victoria’s population.',
    'An average of 500,000 daily passengers arrive in Victoria docking in the Inner Harbour.',
    'Victoria is a popular filming location in Canada due to its proximity to Vancouver and a 6% distance location tax credit.',
    'Cricket has been played at Victoria’s Beacon Hill Park since the mid 19th century.',
    'Victoria is home of the world’s largest all-seaplane airline, Harbour Air.',
    'Future Hall of Fame Basketball player Steve Nash spent most of his childhood in Victoria.',
    'Sequim is pronounced as one syllable, like ‘skwim’',
    'The Hood Canal Bridge, connecting the Olympic and Kitsap Peninsulas, is the longest floating bridge in the world located in a saltwater tidal basin.',
    'Franklin D Roosevelt visited Port Angeles in 1937 to help establish Olympic National Park.'
]

export const SHUFFLE = (array) => {
	let currentIndex = array.length, temporaryValue, randomIndex
	while (0 !== currentIndex) {
		randomIndex = Math.floor(Math.random() * currentIndex)
		currentIndex -= 1
		temporaryValue = array[currentIndex]
		array[currentIndex] = array[randomIndex]
		array[randomIndex] = temporaryValue
	}
	return array
}