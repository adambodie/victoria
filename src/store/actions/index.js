import * as types from '../constants';


export function increment(text) {
	return {
		type: types.INCREMENT,
		text
	}
}
export function decrement(text) {
	return {
		type: types.DECREMENT,
		text
	}
}