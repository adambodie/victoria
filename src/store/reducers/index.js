import { combineReducers } from 'redux'
import { BUTTONS, PAGES, LINKS, BANNERS, FEATURES, RESTAURANTS, FACTS, SHUFFLE, INCREMENT, DECREMENT } from '../constants'


export const buttons = (state = BUTTONS) => state
export const pages = (state = PAGES) => state
export const links = (state = LINKS) => state
export const banners = (state = BANNERS) => state
export const features = (state = SHUFFLE(FEATURES)) => state
export const restaurants = (state = RESTAURANTS) => state
export const facts = (state = SHUFFLE(FACTS)) => state

export function counter(state = 3, action) {
    switch (action.type) {
      case INCREMENT:
        if (state === 15) {
            alert('Try Again')
            return state
        } else {
          return state + 1
        }
      case DECREMENT:
          if (state === 0) {
              alert('Try Again')
              return state
          } else {
            return state - 1
          }
      default:
        return state
    }
  }

export const rootReducer = combineReducers({
    buttons,
    pages,
    links,
    banners,
    features,
    restaurants,
    facts,
    counter
})
