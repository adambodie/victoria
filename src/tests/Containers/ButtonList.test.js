import React from 'react';
import renderer from 'react-test-renderer';
import ButtonList from '../../Components/Containers/ButtonList';
import { Provider } from 'react-redux'
import store from '../../store'
import { BrowserRouter as Router } from 'react-router-dom'

describe('ButtonList', () => {
  const buttons = [{ "link": "eat", "logo": "Victoria" }, 
				 { "link": "buy", "logo": "Port Angeles" },
         { "link": "visit", "logo": "Bremerton" }
        ];
  it('renders buttons', () => {
          const tree = renderer.create(<Provider store={store}><Router><ButtonList data={buttons} /></Router></Provider>).toJSON();
          expect(tree).toMatchSnapshot();
      });
});
