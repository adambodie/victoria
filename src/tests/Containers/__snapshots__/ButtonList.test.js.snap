// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonList renders buttons 1`] = `
<div
  className="buttons"
>
  <a
    href="/buy"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-Port Angeles fa-2x homeButtons"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EE3A43",
          "color": "black",
          "justifyContent": "center",
        }
      }
    />
  </a>
  <a
    href="/visit"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-Bremerton fa-2x homeButtons"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EE3A43",
          "color": "black",
          "justifyContent": "center",
        }
      }
    />
  </a>
</div>
`;
